default.nix

```diff
@@ -17,10 +17,11 @@
 , doStrict    ? false
 # Escape the version bounds from the cabal file. You may want to avoid this function.
 , doJailbreak ? false
-, enableSharedExecutables ? true
+, enableSharedExecutables ? false
 , enableSharedLibraries ? true
 , enableStaticLibraries ? false
-, enableDeadCodeElimination ? true
+#  2020-06-02: NOTE: enableDeadCodeElimination = true: On GHC =< 8.8.3 macOS build falls due to https://gitlab.haskell.org/ghc/ghc/issues/17283, so temporarily set default to `false`
+, enableDeadCodeElimination ? false
```

Comment:
What's the downside here? Larger, slower binaries?

You say the change is temporary. What's the plan for re-enabling this?

Reply:
Well, maybe I was approaching it from point of explaining myself. Why I turn it on and then off, and left a note of possible issues if someone turns it on.

I turned it on, just because "it was available to turn on" and "an optimization", so I turned it on without thinking too much.

Dead code elimination in GHC is pretty time-consuming optimization as far as I heard.

CI should have optimizations disabled, to compile faster.

Actually local development by default probably also should have disabled optimization, so dev loops run faster. But also should have/provide the ability to easily enable them.

So probably by default, for development and CI - it should be turned off. And there should be preset provided that produces release/optimized code.

Reply:
Yes, GHC computes lazy evaluation paths, and removes dead code/bytecode, so smaller binary, smaller RAM footprint, and a bit faster binary execution.